pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -136,6 +137,12 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 					// If the finalize/cleanup process fails for an account we don't want to return
 					// we will flag the account with the Failed Reuse condition, and with state = Failed
 
+					// First we want to see if this was an update race condition where the credentials rotator will update the CR while the finalizer is trying to run.  If that's the case, we want to requeue and retry, before outright failing the account.
+					if strings.HasSuffix(err.Error(), "the object has been modified; please apply your changes to the latest version and try again") {
```

Comment:
This should use the k8serr package to see if it's a NotFound err instead of using strings.HasSuffix.